File: algorithm/src/main/java/baekjoon/p01940/Main.java
package baekjoon.p01940;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();

        int[] inputs = new int[n];
        for (int i = 0; i < inputs.length; i++) {
            inputs[i] = sc.nextInt();
        }

        System.out.println(solve(inputs, m));
    }

    private static int solve(int[] inputs, int m) {
        boolean[] check = new boolean[1000001];
        for (int n : inputs) {
            check[n] = true;
        }

        int ret = 0;
        int left = 0;
        int right = m;

        while (left < right) {
            if (check[left] && check[right]) {
                ret++;
            }
            left++;
            right--;
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p01940/Main.kt
package baekjoon.p01940

fun main(args: Array<String>) {
    val n = readLine()!!.toInt()
    val m = readLine()!!.toInt()
    val inputs = readLine()!!.split(" ").map { it.toInt() }
    println(solve(inputs, m))
}

private fun solve(inputs: List<Int>, m: Int): Int {
    val check = BooleanArray(1000001)
    for (input in inputs) {
        check[input] = true
    }

    var ret = 0
    var left = 0
    var right = m

    while (left < right) {
        if (check[left] && check[right]) {
            ret++
        }

        left++
        right--
    }

    return ret
}